script/helperFunctions.sh

```diff
@@ -1537,16 +1522,49 @@ function verifyContract() {
       --verifier sourcify
   fi
 
-  if [ $? -ne 0 ]; then
-    # verification apparently failed
-    warning "[info] $CONTRACT on $NETWORK with address $ADDRESS could not be verified using Sourcify"
-    return 1
-  else
-    # verification successful
+  if [ $? -eq 0 ]; then
     echo "[info] $CONTRACT on $NETWORK with address $ADDRESS successfully verified using Sourcify"
     return 0
+  else
+    warning "[info] $CONTRACT on $NETWORK with address $ADDRESS could not be verified using Sourcify"
+    return 1
   fi
 }
+
+function getEtherscanApiKeyName() {
+  local NETWORK="$1"
+  local TOML_FILE="foundry.toml"
+
+  if [[ -z "$NETWORK" ]]; then
+    echo "Usage: getEtherscanApiKeyName <network>" >&2
+    return 1
+  fi
+
+  # Extract the line with the API key for the given network
+  local KEY_LINE
+  KEY_LINE=$(awk -v net="$NETWORK" '
```

Comment:
Is awk accessible by default on for example NixOs or Windows?

Reply:
ehm....I dont think so. Our pre install script does not address NixOs or Windows either at the moment if I am not mistaken.

At the moment we dont have a need for this so I would leave it as-is until we run into problems.